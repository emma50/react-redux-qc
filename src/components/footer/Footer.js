import React, { useState } from 'react';
// import clsx from 'clsx';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Link from '@material-ui/core/Link';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import RestoreIcon from '@material-ui/icons/Restore';
import FavoriteIcon from '@material-ui/icons/Favorite';
import LocationOnIcon from '@material-ui/icons/LocationOn';

function Copyright() {
  return (
    <div style={{'backgroundColor': '#ede7f6'}}>
      <Typography variant="body2" color="textSecondary" align="center">
        {'Copyright Â© '}
        <Link color="inherit" href="https://material-ui.com/">
          Your Website
        </Link>{' '}
        {new Date().getFullYear()}
        {'.'}
      </Typography>
    </div>
  );
}

function SimpleBottomNavigation() {
  const [value, setValue] = useState(0);
  return (
    <BottomNavigation 
      value={value}
      onChange={(event, newValue) => {
        setValue(newValue);
      }}
      showLabels 
      style={{
        // 'width': 500,
        'display': 'flex',
        'justifyContent': 'center',
        'background': '#ede7f6',     
      }}
    >
      <BottomNavigationAction label="Recents" icon={<RestoreIcon />} />
      <BottomNavigationAction label="Favorites" icon={<FavoriteIcon />} />
      <BottomNavigationAction label="Nearby" icon={<LocationOnIcon />} />
    </BottomNavigation>
    
  );
}

const useStyles = makeStyles((theme) => ({
  cta: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: 0,
    marginBottom: '20px',
    [theme.breakpoints.up('sm')]: {
      marginTop: '-50px',
    },
  },
  ctaButton: {
    fontWeight: 'bold', 
    textTransform: 'uppercase',
    [theme.breakpoints.up('sm')]: {
      fontSize: 'x-large'
    }, 
  },
  root: {
    display: 'flex',
    backgroundColor: '#ede7f6',
    // overflowX: 'hidden',
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
  paper: {
    // maxWidth: 400,
    margin: `${theme.spacing(1)}px`,
    padding: theme.spacing(2),
    border: 'none',
    backgroundColor: '#ede7f6',
  },
  grid: {
    padding: ` 0 0 ${theme.spacing(1)}px 0 `,
  },
  divider: {
    width: '100vw',
  },
  textTransform: {
    textTransform: 'uppercase',
  },
  lineHeight: {
    lineHeight: 2.5,
  },
  // Desktop view
  desktopRoot: {
    flexGrow: 1,
    backgroundColor: '#ede7f6',
    padding: `0 0 ${theme.spacing(2)}px ${theme.spacing(5)}px `,
    textAlign: 'left',
    [theme.breakpoints.down('sm')]: {
      display: 'none',
    },
  },
}));

export default function Footer() {
  const classes = useStyles();

  return (
    <div>
      <div className={classes.cta}>
        <Button variant="contained" color='primary' className={classes.ctaButton}>apply now</Button>
      </div>
      <div className={classes.root}>
        <div className={classes.paper}>
            <Grid item xs className={classes.grid}>
              <Typography className={classes.textTransform}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            </Grid>
            <Divider className={classes.divider}/>
            <Grid item xs className={classes.grid}>
              <Typography className={classes.textTransform}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            </Grid>
            <Divider className={classes.divider}/>
            <Grid item xs className={classes.grid}>
              <Typography className={classes.textTransform}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            </Grid>
            <Divider className={classes.divider}/>
            <Grid item xs>
              <Typography className={classes.textTransform}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
              <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            </Grid>
        </div>
      </div>
      <div className={classes.desktopRoot}>
        <Grid container spacing={3}>
          <Grid item xs={6} sm={3}>
            <Typography className={classes.textTransform}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
          </Grid>
          <Grid item xs={6} sm={3}>
            <Typography className={classes.textTransform}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
          </Grid>
          <Grid item xs={6} sm={3}>
            <Typography className={classes.textTransform}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
          </Grid>
          <Grid item xs={6} sm={3}>
            <Typography className={classes.textTransform}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
            <Typography className={classes.lineHeight}>hhfghfhdghhhdg</Typography>
          </Grid>
        </Grid>
      </div>
      <SimpleBottomNavigation/>
      <Divider/>
      <Copyright/>
    </div>
  )
}
