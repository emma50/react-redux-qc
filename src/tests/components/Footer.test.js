import React from 'react';
import { render, screen } from '@testing-library/react'; 
import renderer from 'react-test-renderer';
import Footer from '../../components/footer/Footer';


test('render Footer', () => {
  render(<Footer/>);
  expect(screen.getByText(/apply now/i)).toBeInTheDocument();
  expect(screen.getByText(/recents/i)).toBeInTheDocument();
  expect(screen.getByText(/favorites/i)).toBeInTheDocument();
  expect(screen.getByText(/nearby/i)).toBeInTheDocument();
})

test('create Footer snapshot', () => {
  const component = renderer.create(
    <Footer/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
