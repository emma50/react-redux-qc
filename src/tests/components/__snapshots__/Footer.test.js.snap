// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create Footer snapshot 1`] = `
<div>
  <div
    className="makeStyles-cta-10"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-ctaButton-11 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        apply now
      </span>
    </button>
  </div>
  <div
    className="makeStyles-root-12"
  >
    <div
      className="makeStyles-paper-13"
    >
      <div
        className="MuiGrid-root makeStyles-grid-14 MuiGrid-item MuiGrid-grid-xs-true"
      >
        <p
          className="MuiTypography-root makeStyles-textTransform-16 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
      </div>
      <hr
        className="MuiDivider-root makeStyles-divider-15"
      />
      <div
        className="MuiGrid-root makeStyles-grid-14 MuiGrid-item MuiGrid-grid-xs-true"
      >
        <p
          className="MuiTypography-root makeStyles-textTransform-16 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
      </div>
      <hr
        className="MuiDivider-root makeStyles-divider-15"
      />
      <div
        className="MuiGrid-root makeStyles-grid-14 MuiGrid-item MuiGrid-grid-xs-true"
      >
        <p
          className="MuiTypography-root makeStyles-textTransform-16 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
      </div>
      <hr
        className="MuiDivider-root makeStyles-divider-15"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <p
          className="MuiTypography-root makeStyles-textTransform-16 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-desktopRoot-18"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
      >
        <p
          className="MuiTypography-root makeStyles-textTransform-16 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
      >
        <p
          className="MuiTypography-root makeStyles-textTransform-16 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
      >
        <p
          className="MuiTypography-root makeStyles-textTransform-16 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
      >
        <p
          className="MuiTypography-root makeStyles-textTransform-16 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
        <p
          className="MuiTypography-root makeStyles-lineHeight-17 MuiTypography-body1"
        >
          hhfghfhdghhhdg
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiBottomNavigation-root"
    style={
      Object {
        "background": "#ede7f6",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label Mui-selected"
        >
          Recents
        </span>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          Favorites
        </span>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          Nearby
        </span>
      </span>
    </button>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    style={
      Object {
        "backgroundColor": "#ede7f6",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://material-ui.com/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Your Website
      </a>
       
      2020
      .
    </p>
  </div>
</div>
`;
